lib/providers/filesystem/index.js

```diff
@@ -275,3 +282,10 @@ FileSystemProvider.prototype.removeFile = function(container, file, cb) {
   var filePath = path.join(this.root, container, file);
   fs.unlink(filePath, cb);
 };
+
+FileSystemProvider.prototype.getProperPath = function(dirPath, rootPath) {
+  if (dirPath.indexOf(rootPath) !== 0) {
+    var errmsg = ('Illegal character');
+    throw new Error(errmsg);
```

Comment:
Mixing `throw err` and `callback(err)` is confusing for consumers of this module/API. Most provider methods use a callback to signal errors, therefore API consumers don't expect the method to throw an error. Since "illegal character error" is sort of an expected situation, it shouldn't crash the app on unhandled errors.

(It does not matter much here, since `getProperPath` will go away in favour of existing `validateName` function, I am writing this comment to educate you for the future.)